stock_picking_group_by_partner_by_carrier/models/stock_picking.py

```diff
@@ -116,52 +115,53 @@ def copy(self, defaults=None):
         ).copy(defaults)
 
     def get_delivery_report_lines(self):
+        """Return the lines that will be on the report.
+
+        If the picking concerns multiple sale order some fake records are
+        inserted to have the sale information as line separators.
+        Otherwise standard records are returned.
+        """
         self.ensure_one()
+        moves = self.move_lines
         if self.state != "done":
-            moves = self.move_lines.filtered("product_uom_qty").sorted(
-                lambda m: m.sale_line_id.order_id
-            )
-            if len(moves.mapped("sale_line_id.order_id")) > 1:
-                sales_and_moves = []
-                for sale, sale_moves in groupby(
-                    moves, lambda m: m.sale_line_id.order_id
-                ):
-                    sales_and_moves.append(
-                        MockedMove(
-                            product_id=False,
-                            description_picking=sale.get_name_for_delivery_line(),
-                            product_uom_qty=0,
-                            product_uom=False,
-                            lot_name="",
-                        )
-                    )
+            moves = moves.filtered("product_uom_qty")
+        moves = moves.sorted(lambda m: m.sale_line_id.order_id)
+
+        if len(moves.mapped("sale_line_id.order_id")) > 1:
+            grouped_moves = groupby(moves, lambda m: m.sale_line_id.order_id)
+            fake_record = {
+                "product_id": 1,
+                "product_uom_qty": 0,
+                "product_uom": self.env.ref("uom.product_uom_unit").id,
+                "company_id": self.env.user.company_id.id,
+                "location_id": self.env.ref("stock.stock_location_output").id,
+                "location_dest_id": self.env.ref("stock.stock_location_output").id,
+            }
+            if self.state != "done":
+                sales_and_moves = self.env["stock.move"]
+                fake_record["name"] = "fake move"
+                for sale, sale_moves in grouped_moves:
+                    line_desc = sale.get_name_for_delivery_line()
+                    fake_record["description_picking"] = line_desc
+                    sales_and_moves |= sales_and_moves.new(fake_record)
```

Comment:
IMO it's always safer to pass a new dictionary since you're not sure that the dictionary pass as args to `new` is not modified by the method..

Reply:
Didn't think of that. But makes perfect sense.
PR updated. Thanks for your review